hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/ReconstructECContainersCommandHandler.java

```diff
@@ -53,6 +61,13 @@ public void handle(SCMCommand command, OzoneContainer container,
         (ReconstructECContainersCommand) command;
     ECReconstructionCommandInfo reconstructionCommandInfo =
         new ECReconstructionCommandInfo(ecContainersCommand);
+    if (supervisor.getQueueSize() > maxQueueSize) {
```

Comment:
For the replication supervisor, I think the limit check should be in supervisor.addTask(). Then we don't need to add this check in two places (Reconstruction and replication handlers).

Reply:
@sodonnel Updated...